Automattic/jetpack
# Fix/install protect from recommendation ux

Fixes a small UI issue when installing the Protect plugin from the recommendations flow

#### Changes proposed in this Pull Request:

* Makes sure the full promise chain is returned to better detect the completion of plugin install for protect.

#### Other information:


#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* On your test site, reset Jetpack Options from the Jetpack dashboard
* Make sure you have the Jetpack Protect plugin deactivated (or not installed)
* From the plugins screen, activate any plugin to enable the conditional recommendation for Jetpack Protect
* Navigate to: `/wp-admin/admin.php?page=jetpack#/recommendations/summary`
* In the "Recommendations Skipped" section, you will see "Jetpack Protect". Click the "Install" button.
![Screen Shot 2022-08-31 at 3 33 26 PM](https://user-images.githubusercontent.com/18016357/187767941-2210d288-2600-4640-8e7a-dd4a2a33a5d8.png)

* What happened before - Jetpack Protect would show under "Recommendations Enabled" for a moment while installing, briefly disappear and then re-appear under "Recommendations Enabled"
* What should happen now - Jetpack Protect should show under "Recommendations Enabled" for the entire install/ activation duration and then switch to an active state when finished:
![Screen Shot 2022-08-31 at 3 33 30 PM](https://user-images.githubusercontent.com/18016357/187767997-062c81e8-633a-4dac-8f9f-f5dc8daf27c5.png)
![Screen Shot 2022-08-31 at 3 41 18 PM](https://user-images.githubusercontent.com/18016357/187768016-a281845b-36d7-4a8e-b1cc-9d71ce520b22.png)
